doc/testing_snitch.md

```diff
@@ -0,0 +1,146 @@
+
+If an approval test fails for a "good" reason (i.e., it is expected to fail because we have intentionally modified the output, or added something that wasn't there before, or fixed a bug), then the "expected" data needs to be updated. To do so, follow the instructions below:
+ - Run the approval tests; they fail.
+ - Copy the content of the `blanked` directory into the `expected` directory.
+ - Review the diff (e.g., `git diff` or through whatever git interface you use).
+ - If the diff is as intended, commit the new `expected` directory.
+ - Run the approval tests again: they should now pass.
+
+
+### Install tests
+
+The purpose of the install tests is to check that _snitch_ can be used as a library, once installed on a destination folder, rather than from the safe confines of the development repository. The tests themselves are not checking deep and complex features; the intention is only to check that the library is installed correctly, is fully standalone, and that other programs can be created by linking against it.
+
+The install tests are located in the `tests/install_tests` directory.
+
+
+## How to run the tests
+
+### Running tests locally
+
+#### Run-time unit tests
+
+With CMake:
+```cmake
+# Using doctest to test snitch:
+cmake --build build --target snitch_runtime_tests_run
+# Using snitch to test itself:
+cmake --build build --target snitch_runtime_tests_self_run
+```
+
+If you don't use CMake: after building the tests, locate the `snitch_runtime_tests` executable in your build directory and simply run it. The current working directory is irrelevant for these tests.
+
+
+#### Compile-time unit tests
+
+Not implemented yet.
+
+
+#### Approval tests
+
+With CMake:
+```cmake
+cmake --build build --target snitch_approval_tests_run
+```
+
+If you don't use CMake: after building the tests, open a terminal window in the `tests/approval_tests` directory, and run the `snitch_approval_tests` executable.
+
+
+#### Install tests
+
+These are a bit cumbersome to run locally, so the recommendation is to let the automated testing pipeline run them for you.
+
+
+### Automated testing pipeline
+
+In case this document is out-of-date or you have any issue running the tests on your platform, please refer to the [CI scripts](../.github/workflows). The commands in these scripts will always be up-to-date, and you can execute them locally to run the tests on your own machine. Otherwise, GitHub will run these CI scripts once you open your PR, and then for each following commit you push to the branch once the PR is open. This consumes some resources, so please be considerate and try to push commits in bulk, once you think the code should pass tests, rather than one-by-one (i.e., avoid the "commit+push" workflow).
+
+The automated testing pipeline will run *all* tests, for *all* supported platforms. It will also includes other checks, such as test coverage checks. It is necessary for this pipeline to pass before a PR can be merged.
```

Comment:
```suggestion
The automated testing pipeline will run *all* tests, for *all* supported platforms. It will also include other checks, such as test coverage checks. It is necessary for this pipeline to pass before a PR can be merged.
```